modules/nuxeo-apidoc-webengine/src/main/java/org/nuxeo/apidoc/browse/Distribution.java

```diff
@@ -266,15 +267,17 @@ public Resource viewDistribution(@PathParam("distributionId") String distributio
             return ctx.newObject(Resource404.TYPE);
         }
 
+        // resolve version-like distribs based only on version names, taking only persistent distributions and not
+        // including aliases
         List<DistributionSnapshot> snaps = getSnapshotManager().listPersistentSnapshots((ctx.getCoreSession()));
         if (distributionId.matches(VERSION_REGEX.toString())) {
             String finalDistributionId = distributionId;
-            return snaps.stream()
-                        .filter(s -> s.getVersion().equals(finalDistributionId))
-                        .findFirst()
-                        .map(distribution -> ctx.newObject(RedirectResource.TYPE, finalDistributionId,
-                                distribution.getKey()))
-                        .orElseGet(() -> ctx.newObject(Resource404.TYPE));
+            Optional<DistributionSnapshot> resolved = snaps.stream()
+                                                           .filter(s -> s.getVersion().equals(finalDistributionId))
+                                                           .findFirst();
+            if (resolved.isPresent()) {
+                return ctx.newObject(RedirectResource.TYPE, finalDistributionId, resolved.get().getKey());
+            }
```

Comment:
You could use `...findFirst().ifPresent(...)` with a lambda.

Reply:
I did not manage to figure out how to avoid returning (and move on to following logics) if not present (?)

Reply:
Hm yes sorry I hadn't read enough of the surrounding context. Current code is fine then.